### ISSUE TITLE
Disallow isinstance of generic expressions

### ISSUE THREAD

**Issue by ddfisher** [python/mypy#2373]
Now that runtime generics are allowed (thanks to #2302), we should prevent people from making calls like:

``` py
isinstance(["foo"], List[int])
```

because it won't behave as expected.  I.e. in this case, it will return True.  (First instance of user confusion appeared in #2369.)

**Comment by ilevkivskyi** [comment]
Latest version of ``typing.py`` raises ``TypeError`` for things like this. Only unsubscripted generics etc. are allowed with instance/class checks.

**Comment by ddfisher** [comment]
But mypy still doesn't give an error!

**Comment by gvanrossum** [comment]
Oops, sorry. Classified as bug for milestone 0.5. (Since it's a runtime error I don't think it's a very big deal if mypy doesn't catch it.)

**Comment by ilevkivskyi** [comment]
This is fixed now, mypy says
```
error: Parameterized generics cannot be used with class or instance checks
```
